### PR TITLE
Areas tests #41

### PR DESCRIPTION
**Pull request by what-the-hub**
**Refs:**

[**Add gameComponents/Areas tests #41**](https://github.com/what-the-hub/vue-game/issues/41)

**Proposed Changes:**

- Cover Areas component with test (100%)

**Additional Information:**

- Edit `package.json` scripts with adding `"test-coverage-vue": "vue-cli-service test:unit --coverage"` for cheking coverage via vue-cle-service

**Screenshots:**

image

### CODE REVIEW

**Review by Kregopaulgue — APPROVED** [review]

File: src/gameComponents/__tests__/Areas.spec.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    expect(wrapper.emitted('calculate-positions')).toHaveLength(2)
+    expect(wrapper.emitted('calculate-positions')).toMatchObject([
+      [{
+        bottomExcellentArea: 0,
+        bottomGoodArea: 0,
+        topExcellentArea: 0,
+        topGoodArea: 0
+      }],
+      []
+    ])
```
Comment (silchencko) [inline src/gameComponents/__tests__/Areas.spec.ts:53]: Nice
Could you explain what this test checks? Do you test that you called  `calculate-positions` event once, but it was called twice? Once it returned  
```
{
        bottomExcellentArea: 0,
        bottomGoodArea: 0,
        topExcellentArea: 0,
        topGoodArea: 0
}
```
and the second time it returned nothing? Once it was called in `mounted` and the second time - `vm.$emit('calculate-positions')`? What differs these two calls?

File: src/gameComponents/__tests__/Areas.spec.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  })
+})
+
+describe('Areas.vue', () => {
```
Comment (silchencko) [inline src/gameComponents/__tests__/Areas.spec.ts:57]: why are here 2 `describe`s with the same name? I'd rather have one general `describe` and another `describe` (or some `describe`s)  inside the general one.

File: src/gameComponents/__tests__/Areas.spec.ts (hunk 1 above)
Comment (what-the-hub) [reply src/gameComponents/__tests__/Areas.spec.ts:53]: or I need to add `expect(wrapper.emitted('calculate-positions')).toHaveLength(1)`?

File: src/gameComponents/__tests__/Areas.spec.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            score: 0,
+            lastStyle: 'good'
+          }
+        }
+      }
+    })
+  })
+
+  it('should update necessary style', async () => {
+    await Vue.nextTick()
```
[the remark this answers is not in the document]
Comment (what-the-hub) [reply src/gameComponents/__tests__/Areas.spec.ts:76]: no, my mistake

**Pull request merged**